packages/contracts-bedrock/src/governance/DelegatesProposalValidator.sol

```diff
@@ -0,0 +1,158 @@
+
+pragma solidity 0.8.15;
+
+import {IDelegatesProposalValidator} from "interfaces/governance/IDelegatesProposalValidator.sol";
+import {IOptimismGovernor} from "interfaces/governance/IOptimismGovernor.sol";
+import {VotingModule} from "src/governance/VotingModule.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IGovernanceToken} from "interfaces/governance/IGovernanceToken.sol";
+
+contract DelegatesProposalValidator is IDelegatesProposalValidator, Ownable {
+    uint256 public minimumVotingPower;
+    IOptimismGovernor public governor;
+    IGovernanceToken public votingToken;
+    
+    mapping(uint256 => ProposalData) private _proposals;
+    
+    uint256 private _proposalCounter;
+
+    constructor(address _owner, IOptimismGovernor _governor, IGovernanceToken _votingToken) {
+        transferOwnership(_owner);
+        governor = _governor;
+        votingToken = _votingToken;
+    }
+
+    /**
+     * @notice Submit a proposal for delegate approval
+     * @param targets Target addresses for proposal calls
+     * @param values ETH values for proposal calls
+     * @param calldatas Function data for proposal calls
+     * @param description Description of the proposal
+     * @param proposalType Type of the proposal
+     * @return proposalId The ID of the submitted proposal
+     */
+    function submitProposal(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description,
+        uint8 proposalType
+    ) external returns (uint256) {
+        uint256 proposalId = ++_proposalCounter;
+        
+        ProposalData storage proposal = _proposals[proposalId];
+        proposal.proposer = msg.sender;
+        proposal.targets = targets;
+        proposal.values = values;
+        proposal.calldatas = calldatas;
+        proposal.description = description;
+        proposal.proposalType = proposalType;
+        proposal.inVoting = false;
+        proposal.remainingApprovalsRequired = 4; // Hardcoded for now, will change with proposalTypes
+        
+        emit ProposalSubmitted(
+            proposalId,
+            msg.sender,
+            targets,
+            values,
+            calldatas,
+            description,
+            proposalType
+        );
+        
+        return proposalId;
+    }
+
+    /**
+     * @notice Approve a proposal (only callable by delegates with sufficient voting power)
+     * @param proposalId The ID of the proposal to approve
+     */
+    function approveProposal(uint256 proposalId) external {
```

Comment:
There are some inconsistencies with the coding style, sometimes you use `_` for params sometimes there are not. We should keep a consistent coding style based on optimism's and ours coding styles. In this case we should have a `_` prefix on every param